latch_cli/services/register/register.py

```diff
@@ -223,41 +257,55 @@ def register(
     if remote:
         print("Connecting to remote server for docker build [alpha]...")
 
-    with TemporarySerialDir(ctx.ssh_client, remote) as td:
+    with contextlib.ExitStack() as stack:
 
-        dockerfile = ctx.pkg_root.joinpath("Dockerfile")
-        build_logs = build_image(ctx, dockerfile)
-        _print_and_save_build_logs(build_logs, ctx.image_tagged, str(pkg_root))
+        td = stack.enter_context(
+            TemporarySerialDir(ssh_client=ctx.ssh_client, remote=remote)
+        )
+        build_and_serialize(
+            ctx, ctx.default_container.image_name, ctx.default_container.dockerfile, td
+        )
+        protos = recursive_list(td)
 
-        serialize_logs, container_id = _serialize_pkg_in_container(ctx, td)
-        _print_serialize_logs(serialize_logs, ctx.image_tagged)
-        exit_status = ctx.dkr_client.wait(container_id)
-        if exit_status["StatusCode"] != 0:
-            raise ValueError(
-                f"Serialization exited with nonzero exit code: {exit_status['Error']}"
+        for task_name, container in ctx.container_map.items():
+            task_td = stack.enter_context(
+                TemporarySerialDir(ssh_client=ctx.ssh_client, remote=remote)
             )
-
-        upload_image_logs = _upload_pkg_image(ctx)
-        _print_upload_logs(upload_image_logs, ctx.image_tagged)
+            try:
+                build_and_serialize(
+                    ctx, container.image_name, container.dockerfile, task_td
+                )
+                new_protos = recursive_list(task_td)
+                try:
+                    split_task_name = task_name.split(".")
+                    task_name = ".".join(split_task_name[split_task_name.index("wf") :])
+                except ValueError as e:
+                    raise ValueError(
+                        f"Unable to match {task_name} to any of the protobuf files in {new_protos}"
```

Comment:
strange error message for being unable to split a task name

Reply:
the purpose of those two lines of code is to match a serialized task with a list of all serialized objects so it can be pulled out and used at registration